tests/test_utils.h

```diff
@@ -10,11 +12,11 @@
 #endif
 
 /* Log function for tests */
-static const char *log_level_str[] = {
+static const char *const log_level_str[] = {
```

Comment:
I'd extract all "add const" changes is separate commit. Now they are scattered across all patches, but not directly related to changes in them.